pkg/controller/common/settings/canonical_config.go

```diff
@@ -22,7 +22,7 @@ import (
 type CanonicalConfig ucfg.Config
 
 // Options are config options for the YAML file. Currently contains only support for dotted keys.
-var Options = []ucfg.Option{ucfg.PathSep("."), ucfg.AppendValues}
+var Options = []ucfg.Option{ucfg.PathSep("."), ucfg.AppendValues, ucfg.EscapePath()}
```

Comment:
Should this generally be enabed or only in Kibana that supports the escape syntax?

Reply:
I'm not sure how to make this specific to Kibana tbh, this `Options` is used in a lot of places, including receivers like `MergeWith` or `HasKeys` ... I can change all these functions so they accept an "optional `Options`". On the other hand I think it should be fine to enable this for other applications, even if they are not compatible with this syntax?

Reply:
Maybe you are right.